service/history/queues/scheduler.go

```diff
@@ -46,29 +53,52 @@ type (
 		TrySubmit(Executable) (bool, error)
 	}
 
-	SchedulerOptions struct {
-		tasks.ParallelProcessorOptions
-		tasks.InterleavedWeightedRoundRobinSchedulerOptions
+	NamespacePrioritySchedulerOptions struct {
+		WorkerCount      dynamicconfig.IntPropertyFn
+		NamespaceWeights dynamicconfig.MapPropertyFnWithNamespaceFilter
 	}
 
-	schedulerImpl struct {
+	NamespacePrioritySchedulerImpl struct {
```

Comment:
do these 2 needs to be exported?

Reply:
Impl no. I will un-export it. I guess I must copied the name from the exported Options when doing replacement.

Options still need to be exported, because initialization is done in another package.